test/vswhere.test/GlobTests.cpp

```diff
@@ -321,4 +321,34 @@ TEST_CLASS(GlobTests)
         Assert::IsTrue(sut.Match(L"C:\\ShouldNotExist\\MSBuild\\Current\\Bin\\MSBuild.exe"));
         Assert::IsTrue(sut.Match(L"C:\\ShouldNotExist\\MSBuild\\Current\\Bin\\amd64\\MSBuild.exe"));
     }
+
+    BEGIN_TEST_METHOD_ATTRIBUTE(Ignores_Trailing_Separator)
+        TEST_WORKITEM(290)
+    END_TEST_METHOD_ATTRIBUTE()
+    TEST_METHOD(Ignores_Trailing_Separator)
+    {
+        Glob sut(L"C:\\ShouldNotExist\\", L"**\\cl.exe");
+
+        Assert::IsTrue(sut.Match(L"C:\\ShouldNotExist\\VC\\Tools\\MSVC\\14.36.32532\\bin\\Hostx64\\x64\\cl.exe"));
+    }
+
+    BEGIN_TEST_METHOD_ATTRIBUTE(Issue290)
+        TEST_WORKITEM(290)
+    END_TEST_METHOD_ATTRIBUTE()
+    TEST_METHOD(Issue290)
+    {
+        Glob sut(L"C:\\ShouldNotExist", L"**\\cl.exe");
+
+#ifdef _DEBUG
+#endif
+
+        Assert::IsFalse(sut.Match(L"C:\\ShouldNotExist\\Common7\\IDE\\CommonExtensions\\Microsoft\\TeamFoundation\\Team Explorer\\Git\\usr\\bin\\getfacl.exe"));
```

Comment:
This test fails right now, right? Or am I reading the issue wrong?

Reply:
Yes. This is me doing TDD, but after a few other attempts to fix this (and figuring out C++ 11 STL doesn't regex support lookbehind) I'm outside my timebox for now.

See my comment on the PR to you and @matcav about what I think is a path forward. We could try to find a DOS-compatible + globstar implementation, but writing one may be the best option. From a cursory search I found nothing that looks good.